plugins/azure-sites-backend/src/service/router.ts

```diff
@@ -52,28 +66,116 @@ export async function createRouter(
     '/:subscription/:resourceGroup/:name/start',
     async (request, response) => {
       const { subscription, resourceGroup, name } = request.params;
-      console.log('starting...');
-      response.json(
-        await azureSitesApi.start({
-          subscription,
-          resourceGroup,
-          name,
-        }),
+      const token = getBearerTokenFromAuthorizationHeader(
+        request.header('authorization'),
       );
+      const entityRef = request.body.entityRef;
+      if (typeof entityRef !== 'string') {
+        throw new InputError('Invalid entityRef, not a string');
+      }
+      const entity = await catalogApi.getEntityByRef(entityRef, { token });
```

Comment:
So this is a breaking change right? You now need to provide an `entityRef` to be able to use this plugin? I'm not very familiar with this plugin, but does this sound like a sensible requirement?

Reply:
@awanlin I don't suppose you have any experience with `azure-sites`? Or know someone that can reason about this change if it makes sense?

Reply:
Not exactly!, this `entityRef` is passed from the frontend API client. The Integrator no need to pass this explicitly in the plugin. 

https://github.com/deepan10/backstage/blob/514a17d8e6a1dc867815e3cdc52c0993c7bc79d4/plugins/azure-sites/src/api/AzureSitesApiBackendClient.ts#L48-L51